src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -11,6 +13,19 @@ public AsyncRequestProcessor(Executor executor) {
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        Map<String, UserData> userDataMap = new ConcurrentHashMap<>();
+
+        if (userDataMap.containsKey(userId)) {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw new RuntimeException();
+            }
+            return CompletableFuture.supplyAsync(() -> userDataMap.get(userId));
+        } else {
+            userDataMap.put(userId, new UserData(userId, "info about user with id: " + userId));
+        }
+
+        return CompletableFuture.supplyAsync(() -> userDataMap.get(userId));
```

Comment:
The `userDataMap` is being instantiated inside the `processRequest` method which means a new map is created with each request. This defeats the purpose of using a `ConcurrentHashMap` since the map should be shared across requests to be effective. Consider moving `userDataMap` to be a class-level variable so that it is not re-instantiated on each method call.